src/NuGet.Core/NuGet.Build.Tasks.Console/ItemAdapter.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Build.Execution;
+using NuGet.Commands.Restore;
+
+namespace NuGet.Build.Tasks.Console
+{
+    internal class ItemAdapter : IItem
+    {
+        private ProjectItemInstance _item;
+
+        public ItemAdapter(ProjectItemInstance item)
+        {
+            _item = item;
+        }
+
+        public string Identity => _item.EvaluatedInclude;
+
+        public string GetMetadata(string name)
+        {
+            var value = _item.GetMetadataValue(name).Trim();
```

Comment:
Trim might throw if you get a null here.

Reply:
from [API docs for `ProjectItemInstance.GetMetadataValue`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.execution.projectiteminstance.getmetadatavalue?view=msbuild-17-netcore) (emphasis mine)

> Returns
> [String](https://learn.microsoft.com/en-us/dotnet/api/system.string)
> The value of the metadata with the given name, including metadata inherited from item definitions, or **an empty string if no metadata exists with that name**.